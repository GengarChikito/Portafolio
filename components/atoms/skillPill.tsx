import { Tag, theme } from "antd";

/**
 * Defino las 'props' para este componente. Súper simple:
 * - 'text': La habilidad que quiero mostrar (ej. "React", "Python").
 */
export interface SkillPillProps {
    text: string;
}

/**
 * Este es mi átomo 'SkillPill'. Lo uso en la sección Hero
 * para mostrar la lista de mis habilidades.
 */
export function SkillPill({ text }: SkillPillProps) {
    // Aquí está la magia: uso el hook 'useToken' de Ant Design.
    // Esto me da acceso a las variables de color del tema que esté activo (claro u oscuro).
    const { token } = theme.useToken();

    // Saco los colores específicos que quiero del 'token'.
    // Usamos colorBgContainer o colorFillQuaternary para el fondo de la burbuja.
    const bg = token.colorBgContainer || token.colorFillQuaternary;
    // 'colorTextSecondary' es el color de texto secundario (un gris suave).
    const fg = token.colorTextSecondary;
    // 'colorBorderSecondary' es el borde sutil. Si no existe, usa el borde normal.
    const border = token.colorBorderSecondary || token.colorBorder;

    return (
        // Uso el componente <Tag> de Ant Design.
        // Le paso los colores que saqué del 'token' a la prop 'style'.
        // Así, cuando el tema cambie de claro a oscuro, 'bg', 'fg', y 'border'
        // tendrán valores diferentes, y la Tag se repintará sola.
        <Tag
            style={{
                backgroundColor: bg,
                color: fg,
                border: `1px solid ${border}`,
                fontWeight: 500,
                fontSize: 13,
                padding: "4px 10px",
                borderRadius: 9999, // La forma clásica de hacer un "pill"
                marginBottom: 8, // Un pequeño margen para que respiren
            }}
        >
            {text}
        </Tag>
    );
}
