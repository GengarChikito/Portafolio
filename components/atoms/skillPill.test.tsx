// components/atoms/skillPill.test.tsx
import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import React from "react";
import { SkillPill } from "./skillPill";


const tagSpy = vi.fn();

vi.mock("antd", async () => {
    const React = await import("react");
    return {
        Tag: (props: any) => {
            tagSpy(props);
            const { children, ...rest } = props;
            return (
                <div data-testid="antd-tag" {...rest}>
                    {children}
                </div>
            );
        },
        theme: {
            useToken: () => ({
                token: {
                    colorFillQuaternary: "#f5f5f5",
                    colorTextSecondary: "#333333",
                    colorBorderSecondary: "#d9d9d9",
                },
            }),
        },
    };
});


describe("skillPill", () => {
    it("renderiza el texto", () => {
        render(<SkillPill text="React" />);
        expect(screen.getByTestId("antd-tag")).toHaveTextContent("React");
    });

    it("aplica estilos derivados del tema (bg, color, borde) vÃ­a props.style", () => {
        render(<SkillPill text="React" />);
        expect(tagSpy).toHaveBeenCalled();

        const firstCallProps = tagSpy.mock.calls[0][0] as { style: React.CSSProperties };
        const st = firstCallProps.style;

        expect(st.backgroundColor).toBe("#f5f5f5");
        expect(st.color).toBe("#333333");
        expect(st.border).toBe("1px solid #d9d9d9");
        expect(st.fontWeight).toBe(500);
        expect(st.fontSize).toBe(13);
        expect(st.borderRadius).toBe(9999);
        expect(st.padding).toBe("4px 10px");
    });

    it("snapshot", () => {
        const { container } = render(<SkillPill text="React" />);
        expect(container).toMatchSnapshot();
    });
});
