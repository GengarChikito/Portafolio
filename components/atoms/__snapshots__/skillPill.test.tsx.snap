// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`skillPill > snapshot 1`] = `
<div>
  <div
    data-testid="antd-tag"
    style="background-color: rgb(245, 245, 245); color: rgb(51, 51, 51); font-weight: 500; font-size: 13px; padding: 4px 10px; border-radius: 9999px; margin-bottom: 8px;"
  >
    React
  </div>
</div>
`;
